starter-code/scripts/article.js

```diff
@@ -3,28 +3,26 @@
 var articles = [];
 
 function Article (rawDataObj) {
-  // TODO: Use the JS object passed in to complete this constructor function:
-  // Save ALL the properties of `rawDataObj` into `this`
+  this.title = rawDataObj.title;
+  this.category = rawDataObj.category;
+  this.author = rawDataObj.author;
+  this.authorUrl = rawDataObj.authorUrl;
+  this.publishedOn = rawDataObj.publishedOn;
+  this.body = rawDataObj.body;
 }
 
 Article.prototype.toHtml = function() {
   var $newArticle = $('article.template').clone();
-  /* TODO: This cloned article still has a class of template.
-  However, in our modules.css stylesheet, we gave all elements
-  with a class of template a display of none. Let's make
-  sure we're not accidentally hiding our cloned article! */
+  $newArticle.toggleClass('template');
 
   if (!this.publishedOn) $newArticle.addClass('draft');
   $newArticle.data('category', this.category);
 
-  /* TODO: Now use jQuery traversal and setter methods to fill in the rest
-  of the current template clone with properties from this particular Article instance.
-  We need to fill in:
-    1. author name,
-    2. author url,
-    3. article title,
-    4. article body, and
-    5. publication date. */
+  console.log($newArticle)
+  $newArticle.find('h1').text(this.title);
+  $newArticle.find('.article-body').html(this.body);
+  $newArticle.find('.byline address a').text(this.author).attr( 'href', this.authorUrl);
+  $newArticle.find('.article-body').append('<p class=\'pubDate\'>' + this.publishedOn + '</p>');
```

Comment:
There's a `datetime="2000-01-01"` placeholder that should be filled in: https://github.com/bbeeksma/02-jquery-and-the-dom/blob/9b1345f07f515f584a69239dff3d4236b216f5f6/starter-code/index.html#L39